script/constructron.lua

```diff
@@ -711,11 +716,26 @@ me.conditions = {
             local request = constructron.get_vehicle_logistic_slot(i)
             if request then
                 if not (((trunk[request.name] or 0) >= request.min) and ((trunk[request.name] or 0) <= request.max)) then
-                    return false -- condition is not met
+                    -- If fullfillment timout isnt enabled return false
+                    if (global.logistics_fullfillment_timeout == -60) then
+                        return false
+                    end
+
+                    -- if the item isnt in the optimistic_logistics_items
+                    if not custom_lib.table_has_value(global.optimistic_logistics_items, request.name) then
+                        return false -- condition is not met
+                    end
                 end
             end
         end
 
+        -- If we get here it means that we have gotten all the required items and possibly all of the optimistic items
+        -- If the timeout is enabled and the timer hasnt expanded we return false to keep waiting
+        if (timer < global.logistics_fullfillment_timeout and global.logistics_fullfillment_timeout ~= -60) then
+            return false
+        end

```

Comment:
why is this needed? this would result in ctrons always waiting until they reach the timeout.

Reply:
that is correct no? because its optimistic items so it will not leave until it has ALL the items it NEEDs then itll time out at if it hasnt got all the optional ones.

Reply:
it would also cause them to wait until the timeout even when all requests are fulfilled

Reply:
ah, thats a really good point, we could modify the condition for that quite easily.